### PR TITLE
Misc cleanup for treeline

### PR DESCRIPTION
**Pull request by epage**
Separating these out from my work for #4 but wanting to make sure I am building on a solid base.

### CODE REVIEW

**Comment by epage** [comment]
@softprops since it looks like [my crates are the only public clients](https://crates.io/crates/treeline/reverse_dependencies), would you like me to take over maintenance?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        rust: ["stable"]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
```
Comment (softprops) [inline .github/workflows/ci.yml]: Could we use https://github.com/hecrj/setup-rust-action instead? I tend to prefer this as it's more inline with other gh setup-xxxlang-actions.

**Review by softprops — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      with:
+        toolchain: ${{ matrix.rust }}
+        profile: minimal
+        override: true
+    - uses: Swatinem/rust-cache@v1
```
Comment (softprops) [inline .github/workflows/ci.yml]: Just being mindful of bringing in new dependencies. I like the thoughtfulness here but what’s the impact of this on a project that doesn’t have any dependencies? 

If caching provides a significant build impact for this project generally prefer using the official gh actions cache action with cargos official recommendation https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: CI
+on:
+  pull_request:
+    paths:
+    - '**'
+    - '!/*.md'
+    - '!/docs/**'
```
Comment (softprops) [inline .github/workflows/ci.yml]: I think this my have been a copy past error. This folder does not exist

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - "!/LICENSE-*"
```
Comment (softprops) [inline .github/workflows/ci.yml]: This pattern does not match the LICENSE file in this repo

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  push:
+    branches:
+    - master
+    paths:
+    - '**'
+    - '!/*.md'
+    - '!/docs/**'
```
Comment (softprops) [inline .github/workflows/ci.yml]: Same as above. These are likely copy paste errors

**Review by softprops — CHANGES_REQUESTED** [review]
The code changes look good. I think there is some clean up useful for ci. Thoughts on splitting the code changes from the move to gh actions into two separate pulls to move this forward?

if you’d like I could grab the actions bits. I have a standard actions recipe intend to use which works well for me

**Comment by epage** [comment]
> if you’d like I could grab the actions bits. I have a standard actions recipe intend to use which works well for me

Feel free.  I was just applying a subset of my standard template (which is why some things were superfluous, consistency for me was easier for maintaining than customizing exactly to each situation).

File: .github/workflows/ci.yml (hunk 4 above)
Comment (epage) [reply .github/workflows/ci.yml]: Hadn't noticed this is just MIT.  I have no license preference so I just default to the [API guidelines](https://rust-lang.github.io/api-guidelines/necessities.html#c-permissive).  Any thoughts on switching to the combo license?

Comment (softprops) [reply .github/workflows/ci.yml]: I’ll look into it but that feels a bit out of scope for the changes was thinking you were looking to add

**Comment by epage** [comment]
The GH actions have been removed from this PR.

**Review by softprops — APPROVED** [review]

**Pull request merged**